### DISCUSSION TITLE
Can't access database data server side in SvelteKit

### DISCUSSION THREAD

**Discussion by pauldjordan** [supabase/supabase#11896] [Questions] [answered]
I have this code inside +page.server.ts:

```
import { supabase } from "$lib/scripts/supabaseClient"
import type { PageServerLoad } from './$types'

export const load: PageServerLoad = async () => {

    let { data: user, error } = await supabase
        .from('user')
        .select('id')

    console.log(user)
    console.log(error)

}
```

It returns:
```
[]
null
```

But I have this client side code in +page.ts which works perfectly:

```
import { supabase } from "$lib/scripts/supabaseClient"
import type { PageLoad } from './$types';

export const load: PageLoad = async ({params}) => {

    let { data: user, error } = await supabase
        .from('user')
        .select('id')

    console.log(user)
    console.log(error)

}
```

And that returns all the user IDs.

Is there something I'm doing wrong in my server side code?

**Comment by pauldjordan** [comment, marked as answer]
Nevermind. Realised I had enabled RLS to give permissions to logged in users only, which I hadn't realised only works client side. Disabling RLS has fixed it.

**Comment by k3tnalad** [comment]
You made my day, man. Thanks. I've been dealing with the same kinda thing for the last two days, I guess. 
But my thing was rather oblivious. Figured it out by looking at your snippets. 
I was returning `data`, and not the `user` or whatever table is named. Oh my day.

**Reply by pauldjordan** [reply]
Took me a couple of days to figure it out too!  If you want to keep RLS enabled, create a new server side script to invoke the Supabase client with the Supabase secret key, rather than the public key. This will allow the server side code to bypass RLS.

https://kit.svelte.dev/docs/server-only-modules
